src/com/facebook/buck/jvm/kotlin/KotlincToJarStepFactory.java

```diff
@@ -82,21 +83,32 @@ public void createCompileStep(
       ImmutableList.Builder<Step> steps,
       BuildableContext buildableContext) {
 
-    steps.add(
-        new KotlincStep(
-            invokingRule,
-            outputDirectory,
-            sourceFilePaths,
-            pathToSrcsList,
-            ImmutableSortedSet.<Path>naturalOrder()
-                .addAll(
-                    Optional.ofNullable(extraClassPath.apply(buildContext))
-                        .orElse(ImmutableList.of()))
-                .addAll(declaredClasspathEntries)
-                .build(),
-            kotlinc,
-            extraArguments,
-            filesystem));
+    ImmutableSortedSet<Path> kotlinSourceFiles =
+        ImmutableSortedSet.copyOf(
+            sourceFilePaths
+                .stream()
+                .filter(KOTLIN_PATH_MATCHER::matches)
+                .collect(Collectors.toSet()));
```

Comment:
Can you just use `anyMatch` here instead of building a new set? (https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#anyMatch-java.util.function.Predicate-)